pkg/capacity/manager.go

```diff
@@ -89,17 +81,62 @@ func hasPlacementRoles(placementRoles []string) spaceProvisionerConfigPredicate
 	}
 }
 
-func NewClusterManager(namespace string, cl runtimeclient.Client) *ClusterManager {
+func DefaultClusterManager(namespace string, cl runtimeclient.Client) *ClusterManager {
+	return NewClusterManager(namespace, cl, nil)
+}
+
+func NewClusterManager(namespace string, cl runtimeclient.Client, getSpaceCount SpaceCountGetter) *ClusterManager {
 	return &ClusterManager{
-		namespace: namespace,
-		client:    cl,
+		namespace:     namespace,
+		client:        cl,
+		getSpaceCount: getSpaceCount,
 	}
 }
 
 type ClusterManager struct {
-	namespace string
-	client    runtimeclient.Client
-	lastUsed  string
+	getSpaceCount SpaceCountGetter
+	namespace     string
+	client        runtimeclient.Client
+	lastUsed      string
+}
+
+// GetSpaceCountFromSpaceProvisionerConfigs is a function that returns the space count as it is stored in the SpaceProvisionerConfig
+// objects in the provided namespace.
+//
+// NOTE: THIS FUNCTION SHOULD ONLY BE USED IN UNIT TESTS TO SIMPLIFY THEIR SETUP.
```

Comment:
This is also used in tests in other packages. Moving it to the `manager_test.go` would make it not accessible in those other tests.

Reply:
it's minor thing - I'm pointing this out because go build/install will ignore all `*_test.go` files and avoid including packages that are only imported from those test files into the production executable binary.

Could these be a good fit for the `test` folder ? It can be moved in a separate PR though.